tests/unit_tests/json_serialization.cpp

```diff
@@ -0,0 +1,260 @@
+  };
+
+const std::vector<std::vector<std::string>> t_transactions =
+  {
+    {
+      "0100010280e08d84ddcb0106010401110701f254220bb50d901a5523eaed438af5d43f8c6d0e54ba0632eb539884f6b7c02008c0a8a50402f9c7cf807ae74e56f4ec84db2bd93cfb02c2249b38e306f5b54b6e05d00d543b8095f52a02b6abb84e00f47f0a72e37b6b29392d906a38468404c57db3dbc5e8dd306a27a880d293ad0302cfc40a86723e7d459e90e45d47818dc0e81a1f451ace5137a4af8110a89a35ea80b4c4c321026b19c796338607d5a2c1ba240a167134142d72d1640ef07902da64fed0b10cfc8088aca3cf02021f6f655254fee84161118b32e7b6f8c31de5eb88aa00c29a8f57c0d1f95a24dd80d0b8e1981a023321af593163cea2ae37168ab926efd87f195756e3b723e886bdb7e618f751c480a094a58d1d0295ed2b08d1cf44482ae0060a5dcc4b7d810a85dea8c62e274f73862f3d59f8ed80a0e5b9c2910102dc50f2f28d7ceecd9a1147f7106c8d5b4e08b2ec77150f52dd7130ee4f5f50d42101d34f90ac861d0ee9fe3891656a234ea86a8a93bf51a237db65baa00d3f4aa196a9e1d89bc06b40e94ea9a26059efc7ba5b2de7ef7c139831ca62f3fe0bb252008f8c7ee810d3e1e06313edf2db362fc39431755779466b635f12f9f32e44470a3e85e08a28fcd90633efc94aa4ae39153dfaf661089d045521343a3d63e8da08d7916753c66aaebd4eefcfe8e58e5b3d266b752c9ca110749fa33fce7c44270386fcf2bed4f03dd5dadb2dc1fd4c505419f8217b9eaec07521f0d8963e104603c926745039cf38d31de6ed95ace8e8a451f5a36f818c151f517546d55ac0f500e54d07b30ea7452f2e93fa4f60bdb30d71a0a97f97eb121e662006780fbf69002228224a96bff37893d47ec3707b17383906c0cd7d9e7412b3e6c8ccf1419b093c06c26f96e3453b424713cdc5c9575f81cda4e157052df11f4c40809edf420f88a3dd1f7909bbf77c8b184a933389094a88e480e900bcdbf6d1824742ee520fc0032e7d892a2b099b8c6edfd1123ce58a34458ee20cad676a7f7cfd80a28f0cb0888af88838310db372986bdcf9bfcae2324480ca7360d22bff21fb569a530e"
+    }
+  , {
+    }
+  };
+
+// convert hex string to string that has values based on that hex
+// thankfully should automatically ignore null-terminator.
+std::string h2b(const std::string& s)
+{
+  bool upper = true;
+  std::string result;
+  unsigned char val = 0;
+  for (char c : s)
+  {
+    if (upper)
+    {
+      val = 0;
+      if (c <= 'f' && c >= 'a')
+      {
+        val = ((c - 'a') + 10) << 4;
+      }
+      else
+      {
+        val = (c - '0') << 4;
+      }
+    }
+    else
+    {
+      if (c <= 'f' && c >= 'a')
+      {
+        val |= (c - 'a') + 10;
+      }
+      else
+      {
+        val |= c - '0';
+      }
+      result += (char)val;
+    }
+    upper = !upper;
+  }
+  return result;
+}
```

Comment:
This seems to convert "2" to 32, not 2.
You could make it error out on odd length too, wihch would moot that.